src/date-input/internal.tsx

```diff
@@ -41,7 +41,7 @@ const InternalDateInput = React.forwardRef(
       <MaskedInput
         ref={ref}
         {...props}
-        value={isoToDisplay(value)}
+        value={isoToDisplay(value || '')}
```

Comment:
The value is annotated to be a string, however when used in JS environment it is easy to pass `undefined` or `null` which would result in an error on this line.